packages/fluentui/react/src/themes/teams/components/Input/inputStyles.ts

```diff
@@ -49,18 +51,31 @@ const inputStyles: ComponentSlotStylesPrepared<InputProps, InputVariables> = {
     })
   }),
 
-  icon: ({ props: p, variables: v }): ICSSInJSStyle => ({
-    color: v.iconColor,
-    outline: 0,
-    position: v.iconPosition as PositionProperty,
+  icon: ({ props: p, variables: v }): ICSSInJSStyle => {
+    const hasValue = !!p.value && (p.value as string).length !== 0;
```

Comment:
Can we move thing to `getAutoControlledStateFromProps()` for now? And then we can just reuse it in `useStyles`